### PR TITLE
aliases.nix: throw aliases before 2022

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pbsds — APPROVED** [review]
verified the diff with 
```shell
diff \
  <(cat 349431.patch | grep -E '^-  ' | cut -c2- | sd '  (.*) = (.*); #.*' '  $1 = throw "'"'"'$1'"'"' has been renamed to/replaced by '"'"'$2'"'"'"; # Converted to throw 2024-10-17' ) \
  <(cat 349431.patch | grep -E '^\+  ' | cut -c2- )
```

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1232,15 +1232,14 @@ mapAliases {
...
+  packet-cli = throw "'packet-cli' has been renamed to/replaced by 'metal-cli'"; # Converted to throw 2024-10-17
   paperoni = throw "paperoni has been removed, because it is unmaintained"; # Added 2024-07-14
   packet = throw "packet has been removed as it is no longer working and unmaintained"; # Added 2024-03-29
   palemoon = throw "palemoon has been dropped due to python2 being EOL and marked insecure. Use 'palemoon-bin' instead"; # Added 2023-05-18
   pam_usb = throw "'pam_usb' has been removed: abandoned by upstream since 2015."; # Added 2023-10-30
   paper-note = throw "paper-note has been removed: abandoned by upstream"; # Added 2023-05-03
-  paperless = paperless-ngx; # Added 2021-06-06
+  paperless = throw "'paperless' has been renamed to/replaced by 'paperless-ngx'"; # Converted to throw 2024-10-17
   paperless-ng = paperless-ngx; # Added 2022-04-11
-  parity = throw "parity, renamed to openethereum, has been terminated by upstream"; # Added 2020-08-01
```
Comment (pbsds) [inline pkgs/top-level/aliases.nix:1243]: This is the only differing line, which LGTM

**Comment by drupol** [comment]
Oops, this has broken `pkgs.vscode-extensions.rust-lang.rust-analyzer` :(

**Comment by drupol** [comment]
I had to remove it to successfully build my profile, commit at https://github.com/drupol/nixos-x260/commit/f4a3a90d9a3c2368bdba488500312c1163cd3da6

**Comment by pbsds** [comment]
`nix flake check github:drupol/nixos-x260/91e50cb8add8f2e3eff7134904979c0852e9d68a` reveals runs into https://github.com/NixOS/nixpkgs/pull/349783

**Comment by adisbladis** [comment]
Why? We still maintain the same number of attributes, they're just much more unfriendly now.

**Comment by pbsds** [comment]
Would you rather we skip straight to dropping them?

**Comment by AndersonTorres** [comment]
> Would you rather we skip straight to dropping them?

Bad idea imho. It would defeat the purpose of aliases.nix file.

Let's wait one or two releases to drop the aliases.

**Comment by patryk4815** [comment]
image
who will fix unicorn after dropping alias?

**Comment by pbsds** [comment]
* https://github.com/NixOS/nixpkgs/pull/355779

**Pull request merged**